SEED-platform/seed
# redirect home on 404

#### Any background context you want to provide?
When an unknown URL is entered a 404 error is initiated. SEED loads the sidebar, header, and 404 content. With the 404 page loaded the sidebar fails to lookup an active org and prompts the user to create a new one (similar to a user's first time visiting seed). Upon filling out a new organization, the user will be redirected to the home page where they will have a new organization attached to their profile.

#### What's this PR do?
Instead of sending users to a 404 or 500 page, all 404 and 500 errors will redirect to the home page '/' with an alert message in the header area indicating the error and reason.

The backend will redirect to '/' with a query param `?http_error=404` or `500`. The query parameter is visible in the url bar initially but soon disappears as the home page continues to load.

#### How should this be manually tested?
To test locally in docker a few settings need to be modified. 
- /config/settings/docker_dev.py --> change `DEBUG` to `False` 
- /docker-compose.dev.yml --> change `web`...`command:` to  `python3 manage.py runserver --insecure --settings=config.settings.docker_dev 0.0.0.0:80`

Go to any page that does not exist. You should be redirected to "/app/"

#### What are the relevant tickets?
#3140 

#### Screenshots (if appropriate)


https://user-images.githubusercontent.com/58446472/156236768-d06ced06-7dce-4934-95c3-f84ec4ee55b1.mov
